lib/Dialect/TritonGPU/IR/Ops.cpp

```diff
@@ -596,16 +596,12 @@ int32_t LocalAllocOp::getAlignmentOrDefault() {
     return *align;
   }
 
+  if (!isSharedMemoryAlloc())
+    return 8;
+
   auto ty = getType();
-  auto shapePerCTA = triton::gpu::getShapePerCTA(ty);
-  auto bytes =
-      product<int64_t>(shapePerCTA) * (ty.getElementTypeBitWidth() / 8);
-
-  // XXX(Keren): magic numbers 256 and 1024
-  // Software swizzling calculates phase based on offset, while hardware
-  // swizzling do that based on physical address. Thus only by setting the
-  // alignment to 1024 can ensure the correctness.
-  return bytes > 256 ? 1024 : 8;
+  auto enc = cast<SharedEncodingAttr>(ty.getEncoding());
+  return enc.getAlignment(ty.getElementType());
```

Comment:
Taking another look, I noticed that allocations were in fact being aligned because of this heuristic. So, I've just updated that function to look at the encoding.

Reply:
However, I'd like to point out that a memory descriptor can accept not only shared memory encoding.

Reply:
If we associate alignment with the shared encoding, the same attribute or interface has to be added to encoding X, not documented yet.

Reply:
What other encodings are there?

Reply:
Changed to `dyn_cast` with a fallback to 16-byte alignment,